### DISCUSSION TITLE
Accessing composite types

### DISCUSSION THREAD

**Discussion by robhanlon22** [jkk/honeysql#487] [Q&A] [answered]
PostgreSQL supports syntax for extracting fields from composite types.

https://www.postgresql.org/docs/current/rowtypes.html#ROWTYPES-ACCESSING

Here's a toy example:

```sql
SELECT (foos).field from (select foos from foos) f
```

Since the parens are significant here, how would you recommend access this field? This works, but it uses `:raw`, and doesn't handle quoting correctly:

```clojure
(-> (select [[:raw [[:composite :foos]] ".field"]])
    (from [(-> (select :foos) (from :foos)) :f]))
```

which results in something functional and close to the example after formatting:

```sql
SELECT ((foos)).field FROM (SELECT foos FROM foos) AS f
```

Is there a better way to do this? Normally one would use a namespaced keyword, but it won't work here. 

Thanks for your help!

**Comment by seancorfield** [comment]
Support for this was added in 2.4.1002 (early March) -- see #474

**Reply by seancorfield** [reply, marked as answer]
See https://cljdoc.org/d/com.github.seancorfield/honeysql/2.4.1026/doc/getting-started/sql-special-syntax-#dot- in the documentation.

**Comment by robhanlon22** [comment]
Thank you! I actually implemented this dot function myself, glad to see it's already present!
